src/shogun/lib/DynamicArray.h

```diff
@@ -540,10 +645,13 @@ template <class T> class CDynamicArray :public CSGObject
 		}
 
 		/** shuffles the array (not thread safe!) */
-		inline void shuffle() { m_array.shuffle(); }
+		inline void shuffle()
+		{
+			std::random_shuffle(m_array.begin(), m_array.end());
+		}
 
 		/** shuffles the array with external random state */
-		inline void shuffle(CRandom * rand) { m_array.shuffle(rand); }
+		// inline void shuffle(CRandom * rand) { m_array.shuffle(rand); }
```

Comment:
replace this function with:
```
template<class RandomFunc>
inline void shuffle(RandomFunc&& r)
{
std::random_shuffle(m_array.begin(), m_array.end(), r);
}
```
because we want to be able to fix the seed of the shuffle